warehouse/forklift/legacy.py

```diff
@@ -866,7 +890,15 @@ def file_upload(request):
         else {
             name: {
                 "url": url,
-                "verified": _verify_url(url=url, publisher_url=publisher_base_url),
+                "verified": _verify_url_with_trusted_publisher(
+                    url=url,
+                    publisher_url=publisher_base_url,
+                )
+                or _verify_url_pypi(
+                    url=url,
+                    project_name=project.name,
+                    project_normalized_name=project.normalized_name,
+                ),
```

Comment:
I added the helper function. As for refactoring to a separate module/file, I agree, particularly since we'll be adding even more verification options in the future

Reply:
@woodruffw while adding support for the `github.io` URLs, I realised that it might make sense to move the current `_verify_url_with_trusted_publisher` to the `OIDCPublisherMixin` class as a method (since we're verifying the URL against the publisher's base URL, which lives in that class)

Then, for provider-specific verification rules we want to add (like GitHub's `user.github.io/repo`) , we can override that method in the GitHub subclass to add that logic. WDYT?

For illustration, I mean something like this:
```python
class GitHubPublisher(GitHubPublisherMixin, OIDCPublisher)
...
    def verify_url(self, url: str):
        # the parent's `verify_url` contains the logic that is currently in `_verify_url_with_trusted_publisher()`
        if super().verify_url(url):  
            return True

        docs_url = f"https://{self.repository_owner}.github.io/{self.repository_name}"
        return url == docs_url or url.startswith(f"{docs_url}/")
```

Reply:
> Then, for provider-specific verification rules we want to add (like GitHub's `user.github.io/repo`) , we can override that method in the GitHub subclass to add that logic. WDYT?

That seems reasonable to me -- the only hesitation I have is that it makes these OIDC mixins bigger and bigger, but in terms of grouping related components together it makes sense 🙂

Reply:
Given that we might be verifying URLs from a variety of source truths (which may or may not be trusted publishers) I think it's OK as-is for now.